src/main/java/org/opentripplanner/middleware/bugsnag/BugsnagJobs.java

```diff
@@ -17,36 +28,118 @@
+        Scheduler.scheduleJob(
+            new BugsnagEventRequestJob(projectIds),
+            initialDelay,
+            BUGSNAG_EVENT_REQUEST_JOB_DELAY_IN_HOURS,
+            TimeUnit.HOURS);
+    }
+
+    /**
+     * Seed a project if the last event data request is greater than a day.
+     */
+    private static Set<String> seedDataIfNeeded() {
+        Set<String> projectsRequested = new HashSet<>();
+        Set<String> projectIds = MonitoredComponent.getComponentsByProjectId().keySet();
+        projectIds.forEach(projectId ->
+            {
+                int hoursSinceLastRequest = getHoursSinceLastRequest(projectId);
+                // Only seed data if days since is greater than one day (otherwise we're duplicating the scheduled)
+                if (hoursSinceLastRequest > BUGSNAG_EVENT_REQUEST_JOB_DELAY_IN_HOURS) {
+                    LOG.debug("Triggered Bugsnag event request for project {}", projectId);
+                    BugsnagEventRequestJob.triggerEventDataRequestForProject(projectId, hoursSinceLastRequest / 24);
+                    projectsRequested.add(projectId);
+                }
+            }
+        );
+        return projectsRequested;
+    }
+
+    /**
+     * Define how long in hours since a project's last event data request took place. Restrict to
+     * {@link BugsnagDispatcher#BUGSNAG_REPORTING_WINDOW_IN_DAYS} (in hours) if greater than this.
+     */
+    private static int getHoursSinceLastRequest(String projectId) {
+        int hoursSinceLastRequest = 0;
+        // Get latest successful request for project.
+        BugsnagEventRequest compareRequest = getLatestComparisonRequest(projectId);
+        if (compareRequest != null) {
+            // Otherwise, determine how many hours have passed since the end of the last request's time window.
+            long reportingGapMillis = new Date().getTime() - compareRequest.getTimeWindowEndInMillis();
+            // TimeUnit#convert truncates/rounds down, so add one to account for partial hours.
+            long hours = TimeUnit.HOURS.convert(reportingGapMillis, TimeUnit.MILLISECONDS);
+            hoursSinceLastRequest = Math.toIntExact(hours) + 1;
+        }
+        // Restrict the hours since last request if greater than the reporting window.
+        return Math.max(hoursSinceLastRequest, BUGSNAG_REPORTING_WINDOW_IN_DAYS * 24);
+    }
+
+    /**
+     * Retrieve the latest event request for a project.
+     */
+    private static BugsnagEventRequest getLatestComparisonRequest(String projectId) {
+        BugsnagEventRequest lastCompleted = getLatestCompletedRequestForProject(projectId);
+        BugsnagEventRequest lastIncomplete = getLatestIncompleteRequestForProject(projectId);
+        if (lastIncomplete == null) return lastCompleted;
+        boolean lastIncompleteIsExpired = lastIncomplete.status.equalsIgnoreCase("expired");
+        if (lastCompleted == null) return lastIncompleteIsExpired ? null : lastIncomplete;
```

Comment:
Can you add some inline comments describing what each of these items is checking for?